src/coreclr/jit/unwindriscv64.cpp

```diff
@@ -20,8 +20,209 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 #if defined(FEATURE_CFI_SUPPORT)
 short Compiler::mapRegNumToDwarfReg(regNumber reg)
 {
-    NYI_RISCV64("mapRegNumToDwarfReg-----unimplemented on RISCV64 yet----");
-    return 0;
+    short dwarfReg = DWARF_REG_ILLEGAL;
+
+    switch (reg)
+    {
+        case REG_R0:
+            dwarfReg = 0;
+            break;
+        case REG_RA:
+            dwarfReg = 1;
+            break;
```

Comment:
I can make something not depending on emitter methods like:

```C++
// On RISC-V registers from R0 to F31 
// can be mapped directly to dwarf structure
if (reg >= REG_R0 && reg <= REG_F31)
{
    dwarfReg = static_cast<short>(reg);
}
else
{
    NYI("CFI codes"); // e.g. V-extension
}
```

but I'm not sure if I should, because the rest architectures implements that function using `switch`

Reply:
> I can make something not depending on emitter methods

They are static helpers so it's not a big deal IMO, they are already used in Compiler::unwindSaveReg and in a few places in unwindarm64. But what you suggested is also ok.